src/__tests__/cleanup.js

```diff
@@ -27,11 +27,11 @@ test('cleanup does not error when an element is not a child', async () => {
   await cleanup()
 })
 
-test('cleanup waits for queued microtasks during unmount sequence', async () => {
+test('cleanup runs effect cleanup functions', async () => {
   const spy = jest.fn()
 
   const Test = () => {
-    React.useEffect(() => () => setImmediate(spy))
+    React.useEffect(() => spy)
```

Comment:
All we can guarantee is that the cleanup function is called, so that's all we'll test.

Reply:
:+1: The problem is that we can't know what kind of task is queued from the cleanup function. We only supported `setImmediate` but any longer running task wasn't caught as far as I can tell.